src/hooks/useDelegates.ts

```diff
@@ -0,0 +1,22 @@
+import useSafeInfo from '@/hooks/useSafeInfo'
+import useWallet from '@/hooks/wallets/useWallet'
+import { useGetDelegatesQuery } from '@/store/gateway'
+import { skipToken } from '@reduxjs/toolkit/query/react'
+
+const useDelegates = () => {
+  const {
+    safe: { chainId },
+    safeAddress,
+  } = useSafeInfo()
+
+  return useGetDelegatesQuery(chainId && safeAddress ? { chainId, safeAddress } : skipToken)
```

Comment:
Is this cached?

Reply:
Yes, the data is cached and updates only if `chainId` or `safeAddress` changes. Even if we have `useDelegates` in multiple components it will still access the cached data instead of making copies.